contracts/erc20/contracts/src/v08/LibERC20TokenV08.sol

```diff
@@ -0,0 +1,153 @@
+    ///      Reverts if the return data is invalid or the call reverts.
+    /// @param token The address of the token contract.
+    /// @param spender The address that receives an allowance.
+    /// @param allowance The allowance to set.
+    function compatApprove(IERC20TokenV08 token, address spender, uint256 allowance) internal {
+        bytes memory callData = abi.encodeWithSelector(token.approve.selector, spender, allowance);
+        _callWithOptionalBooleanResult(address(token), callData);
+    }
+
+    /// @dev Calls `IERC20TokenV08(token).approve()` and sets the allowance to the
+    ///      maximum if the current approval is not already >= an amount.
+    ///      Reverts if the return data is invalid or the call reverts.
+    /// @param token The address of the token contract.
+    /// @param spender The address that receives an allowance.
+    /// @param amount The minimum allowance needed.
+    function approveIfBelow(IERC20TokenV08 token, address spender, uint256 amount) internal {
+        if (token.allowance(address(this), spender) < amount) {
+            compatApprove(token, spender, type(uint256).max);
+        }
+    }
+
+    /// @dev Calls `IERC20TokenV08(token).transfer()`.
+    ///      Reverts if the return data is invalid or the call reverts.
+    /// @param token The address of the token contract.
+    /// @param to The address that receives the tokens
+    /// @param amount Number of tokens to transfer.
+    function compatTransfer(IERC20TokenV08 token, address to, uint256 amount) internal {
+        bytes memory callData = abi.encodeWithSelector(token.transfer.selector, to, amount);
+        _callWithOptionalBooleanResult(address(token), callData);
+    }
+
+    /// @dev Calls `IERC20TokenV08(token).transferFrom()`.
+    ///      Reverts if the return data is invalid or the call reverts.
+    /// @param token The address of the token contract.
+    /// @param from The owner of the tokens.
+    /// @param to The address that receives the tokens
+    /// @param amount Number of tokens to transfer.
+    function compatTransferFrom(IERC20TokenV08 token, address from, address to, uint256 amount) internal {
+        bytes memory callData = abi.encodeWithSelector(token.transferFrom.selector, from, to, amount);
+        _callWithOptionalBooleanResult(address(token), callData);
+    }
+
+    /// @dev Retrieves the number of decimals for a token.
+    ///      Returns `18` if the call reverts.
+    /// @param token The address of the token contract.
+    /// @return tokenDecimals The number of decimals places for the token.
+    function compatDecimals(IERC20TokenV08 token) internal view returns (uint8 tokenDecimals) {
+        tokenDecimals = 18;
+        (bool didSucceed, bytes memory resultData) = address(token).staticcall(DECIMALS_CALL_DATA);
+        if (didSucceed && resultData.length >= 32) {
+            tokenDecimals = uint8(LibBytesV08.readUint256(resultData, 0));
```

Comment:
I don't understand the reasoning behind using `LibBytesV08` here. Solidity can do this conversion for you
```suggestion
            tokenDecimals = abi.decode(resultData, (uint8));
```